packages/discord.js/src/structures/GuildOnboarding.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const Base = require('./Base');
+const { GuildOnboardingPrompt } = require('./GuildOnboardingPrompt');
+
+/**
+ * Represents the onboarding data of a guild.
+ * @extends {Base}
+ */
+class GuildOnboarding extends Base {
+  constructor(client, data) {
+    super(client);
+
+    const guild = this.guild;
+
+    /**
+     * The id of the guild this onboarding data is for
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * The prompts shown during onboarding
+     * @type {Collection<Snowflake, GuildOnboardingPrompt>}
+     */
+    this.prompts = data.prompts.reduce((prompts, prompt) => {
+      prompts.set(prompt.id, new GuildOnboardingPrompt(client, prompt, this.guildId));
+      return prompts;
+    }, new Collection());
+
+    /**
+     * The ids of the channels that new members get opted into automatically
+     * @type {Collection<Snowflake, GuildChannel>}
+     */
+    this.defaultChannels = data.default_channels.reduce((channels, channelId) => {
+      channels.set(channelId, guild.channels.cache.get(channelId));
+      return channels;
+    }, new Collection());
```

Comment:
Channels may not be cached. This should be implemented like guild member roles.
https://github.com/discordjs/discord.js/blob/0e4224be36f6e64ccd8366f6cc86e6ce3f33eec5/packages/discord.js/src/managers/GuildMemberRoleManager.js#L35

Reply:
```suggestion
    this.defaultChannels = data.default_channels.reduce((channels, channelId) =>
      channels.set(channelId, guild.channels.cache.get(channelId)),
      new Collection()
    );
```
.set returns the collection so we don't need to make this function unnessercarily longer